redfish-core/lib/chassis.hpp

```diff
@@ -413,6 +413,42 @@ inline void requestRoutesChassis(App& app)
                                 locationInterface, "LocationCode");
                         }
 
+                        const std::string versionInterface =
+                            "xyz.openbmc_project.Software.Version";
+                        if (std::find(interfaces2.begin(), interfaces2.end(),
+                                      versionInterface) != interfaces2.end())
+                        {
+                            crow::connections::systemBus->async_method_call(
+                                [asyncResp, chassisId(std::string(chassisId))](
+                                    const boost::system::error_code ec,
+                                    const std::variant<std::string>& property) {
+                                    if (ec)
+                                    {
+                                        BMCWEB_LOG_DEBUG
+                                            << "DBUS response error for "
+                                               "Version";
+                                        messages::internalError(asyncResp->res);
+                                        return;
+                                    }
+
+                                    const std::string* value =
+                                        std::get_if<std::string>(&property);
+                                    if (value == nullptr)
+                                    {
+                                        BMCWEB_LOG_DEBUG
+                                            << "Null value returned "
+                                               "for Version";
+                                        messages::internalError(asyncResp->res);
+                                        return;
+                                    }
+                                    asyncResp->res
+                                        .jsonValue["FirmwareVersion"] = *value;
```

Comment:
The validator should flag this. This is OEM.. it should be under "Oem"...

Reply:
this is old commit Sorry I need to upload new one which has fix